samples/scala-protobuf-validated/src/test/scala/com/example/validated/ValidatedActionIntegrationSpec.scala

```diff
@@ -0,0 +1,40 @@
+package com.example.validated
+
+import akka.grpc.GrpcServiceException
+import kalix.scalasdk.testkit.KalixTestKit
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.concurrent.duration._
+
+// This class was initially generated based on the .proto definition by Kalix tooling.
+//
+// As long as this file exists it will not be overwritten: you can maintain it yourself,
+// or delete it so it is regenerated as needed.
+
+class ValidatedActionIntegrationSpec extends AnyWordSpec with Matchers with BeforeAndAfterAll with ScalaFutures {
+
+  implicit private val patience: PatienceConfig = PatienceConfig(5.seconds, 500.millis)
+
+  private val testKit = KalixTestKit(Main.createKalix()).start()
+
+  private val client = testKit.getGrpcClient(classOf[Validated])
+
+  "ValidatedAction" must {
+
+    "handle command CallMeMaybe" in {
+      val result = client.callMeMaybe(Request("valid.email@example.com")).futureValue
+      result should ===(Response("Hello valid.email@example.com, gosh what a valid email you sent me"))
+
+      // we can't test validation failure because we can't construct an invalid request
```

Comment:
Did you consider enabling http transcoding and then having a http client here to make the invalid request? But then we would need to hardcode the host I guess.. 🤔 Probably not worth the hassle.

Reply:
Yeah, would need to figure out the test container port, which seemed to tricky to be worth it.